src/mlir/ast-utils.cc

```diff
@@ -288,6 +300,15 @@ namespace mlir::verona::ASTInterface
     return args.lock()->nodes[n - 1];
   }
 
+  std::vector<::ast::WeakAst> getAllOperands(::ast::WeakAst ast)
+  {
+    auto numOps = numOperands(ast);
+    std::vector<::ast::WeakAst> ops;
+    for (size_t i = 0; i < numOps; i++)
+      ops.push_back(getOperand(ast, i));
+    return ops;
+  }
+
```

Comment:
I'm a bit nervous about embedding `std::vector`s at this kind of level.  Could this be written as a template that takes a reference to the collection as a parameter, something like:
```suggestion
  template<class T>
  std::vector<::ast::WeakAst> getAllOperands(T &ops, ::ast::WeakAst ast)
  {
    auto numOps = numOperands(ast);
    for (size_t i = 0; i < numOps; i++)
      ops.push_back(getOperand(ast, i));
    return ops;
  }
```

Reply:
Sorry, this is a commit from a different PR, in which this change is based on (#195). I see what you mean and will look at the other cases, too, but will update the other PR with it, then rebase this one. This is for the three last commits only.

I don't know how to do multiple reviews in one without carrying the commits along in Github.